chalk-solve/src/coherence/solve.rs

```diff
@@ -268,11 +261,10 @@ impl<I: Interner> CoherenceSolver<'_, I> {
     }
 }
 
-fn params<'a, I: Interner>(interner: &I, impl_datum: &'a ImplDatum<I>) -> &'a [Parameter<I>] {
-    impl_datum
-        .binders
-        .value
-        .trait_ref
-        .substitution
-        .parameters(interner)
+fn params<'a, I: Interner>(
```

Comment:
This is such a short function now. And since we have that safety comment about skipping binders, let's just inline this where it's used.